src/main/java/org/gridsuite/boundary/importer/job/CgmesBoundaryServiceRequester.java

```diff
@@ -0,0 +1,120 @@
+                .build();
+
+        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+        return response.statusCode() == 200;
+    }
+
+    private HttpRequest.BodyPublisher ofMimeMultipartData(Map<Object, Object> data,
+                                                         String boundary) {
+        // Result request body
+        List<byte[]> byteArrays = new ArrayList<>();
+
+        // Separator with boundary
+        byte[] separator = ("--" + boundary + "\r\nContent-Disposition: form-data; name=").getBytes(StandardCharsets.UTF_8);
+
+        // Iterating over data parts
+        for (Map.Entry<Object, Object> entry : data.entrySet()) {
+
+            // Opening boundary
+            byteArrays.add(separator);
+
+            // If value is type of Path (file) append content type with file name and file binaries, otherwise simply append key=value
+            if (entry.getValue() instanceof TransferableFile) {
+                var file = (TransferableFile) entry.getValue();
+                String mimeType = "application/octet-stream";
+                byteArrays.add(("\"" + entry.getKey() + "\"; filename=\"" + file.getName()
+                        + "\"\r\nContent-Type: " + mimeType + "\r\n\r\n").getBytes(StandardCharsets.UTF_8));
+                byteArrays.add(file.getData());
+                byteArrays.add("\r\n".getBytes(StandardCharsets.UTF_8));
+            } else {
+                byteArrays.add(("\"" + entry.getKey() + "\"\r\n\r\n" + entry.getValue() + "\r\n")
+                        .getBytes(StandardCharsets.UTF_8));
+            }
+        }
+
+        // Closing boundary
+        byteArrays.add(("--" + boundary + "--").getBytes(StandardCharsets.UTF_8));
+
+        // Serializing as byte array
+        return HttpRequest.BodyPublishers.ofByteArrays(byteArrays);
+    }
+
+    public List<String> getBoundariesIds() throws IOException, InterruptedException {
+        try {
+            HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(serviceUrl + API_VERSION + "/boundaries/ids"))
+                .GET()
+                .build();
+
+            HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+            if (response.statusCode() == 200) {
+                String json = response.body();
+
+                List<String> result = new ArrayList<>();
+                JSONArray array = new JSONArray(json);
+                for (int i = 0; i < array.length(); i++) {
+                    result.add(array.getString(i));
+                }
+                return result;
+            }
+        } catch (IOException e) {
+            LOGGER.error("I/O Error while getting all boundary ids");
+        } catch (InterruptedException e) {
+            LOGGER.error("Interruption while getting all boundary ids");
+            Thread.currentThread().interrupt();
```

Comment:
Why this interrupt ?

Reply:
To fix sonarcloud report :  "InterruptedException" should not be ignored